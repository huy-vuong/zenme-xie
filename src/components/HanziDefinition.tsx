import PropTypes from 'prop-types';
import React from 'react';
import styles from 'zenme-xie/components/HanziDefinition.module.scss';

export default function HanziDefinition({ definition }: HanziDefinitionProp) {
  return <div className={styles.definition}>{definition}</div>;
}

interface HanziDefinitionProp {
  definition?: string;
}

HanziDefinition.propTypes = {
  definition: PropTypes.string,
};
