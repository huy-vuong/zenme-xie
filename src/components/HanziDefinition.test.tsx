import { render } from '@testing-library/react';
import React from 'react';
import HanziDefinition from 'zenme-xie/components/HanziDefinition';

test('renders the given definition', () => {
  const definition = 'kittykat';
  const { getByText, asFragment } = render(
    <HanziDefinition definition={definition} />
  );
  getByText(definition);
  expect(asFragment()).toMatchSnapshot();
});

test('renders nothing given no pinyin', () => {
  const { asFragment } = render(<HanziDefinition />);
  expect(asFragment()).toMatchSnapshot();
});
