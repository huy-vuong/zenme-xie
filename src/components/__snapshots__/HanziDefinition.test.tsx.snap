// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders nothing given no pinyin 1`] = `
<DocumentFragment>
  <div
    class="definition"
  />
</DocumentFragment>
`;

exports[`renders the given definition 1`] = `
<DocumentFragment>
  <div
    class="definition"
  >
    kittykat
  </div>
</DocumentFragment>
`;
